src/protocol/data/memcache/parse.c

```diff
@@ -782,31 +790,93 @@ _parse_req_hdr(struct request *req, struct buf *buf)
 parse_rstatus_t
 parse_req(struct request *req, struct buf *buf)
 {
-    parse_rstatus_t status = PARSE_EUNFIN;
+    parse_rstatus_t status = PARSE_OK;
     char *old_rpos = buf->rpos;
+    bool leftmost = (buf->rpos == buf->begin);
+
+    /*
+     * we allow partial value in the request (but not the head portion),
+     * so that we can incrementally fill in a large value over multiple socket
+     * reads. This is more useful for the server which allows more predictable
+     * buffer management (e.g. no unbounded read buffer). Currently partial
+     * value is not implemented for the response.
+     */
+    switch (req->rstate) {
+    case REQ_PARSING: /* a new request */
+        log_verb("parsing buf %p into req %p", buf, req);
+        req->first = true;
+        status = _parse_req_hdr(req, buf);
+        if (status == PARSE_EUNFIN) {
+            log_verb("incomplete data: reset read position, jump back %zu bytes",
+                    buf->rpos - old_rpos);
+            /* return and start from beginning next time */
+            request_reset(req);
+            buf->rpos = old_rpos;
```

Comment:
You are right. I created a new test to capture this condition (and without the break it failed). Thanks for catching this!